ttnctl/cmd/devices_export.go

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2019 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/TheThingsNetwork/api"
+	"github.com/TheThingsNetwork/api/handler"
+	"github.com/TheThingsNetwork/go-utils/log"
+	"github.com/TheThingsNetwork/ttn/ttnctl/util"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func exportV3Device(dev *handler.Device, flags *pflag.FlagSet) *v3Device {
+	v3Dev := &v3Device{}
+	v3Dev.IDs.ApplicationIDs.ApplicationID = dev.AppID
+	v3Dev.IDs.DeviceID = dev.DevID
+
+	if dev.Latitude != 0 || dev.Longitude != 0 {
+		v3Dev.Locations = map[string]v3DeviceLocation{
+			"user": {
+				Latitude:  float64(dev.Latitude),
+				Longitude: float64(dev.Longitude),
+				Altitude:  dev.Altitude,
+				Source:    "SOURCE_REGISTRY",
+			},
+		}
+	}
+
+	v3Dev.Name = dev.DevID
+	v3Dev.Description = dev.Description
+	v3Dev.Attributes = dev.Attributes
+
+	lorawanDevice := dev.GetLoRaWANDevice()
+
+	v3Dev.IDs.JoinEUI = lorawanDevice.GetAppEUI().String()
+	v3Dev.IDs.DevEUI = lorawanDevice.GetDevEUI().String()
+	if lorawanDevice.AppKey.String() != "" {
+		v3Dev.RootKeys = &v3DeviceRootKeys{
+			AppKey: v3DeviceKey{Key: lorawanDevice.AppKey.String()},
+		}
+	}
+	if devAddr := lorawanDevice.DevAddr; !devAddr.IsEmpty() {
+		v3Dev.DevAddr = devAddr.String()
+		v3Dev.Session = &v3DeviceSession{}
+		v3Dev.Session.DevAddr = lorawanDevice.DevAddr.String()
+		if lorawanDevice.AppSKey.String() != "" {
+			v3Dev.Session.Keys.AppSKey = v3DeviceKey{Key: lorawanDevice.AppSKey.String()}
+		}
+		if lorawanDevice.NwkSKey.String() != "" {
+			v3Dev.Session.Keys.FNwkSIntKey = v3DeviceKey{Key: lorawanDevice.NwkSKey.String()}
+		}
+		v3Dev.Session.LastFCntUp = lorawanDevice.FCntUp
+		v3Dev.Session.LastNFCntDown = lorawanDevice.FCntDown
+	}
+
+	v3Dev.MACSettings.Supports32BitFCnt = lorawanDevice.Uses32BitFCnt
+	v3Dev.MACSettings.ResetsFCnt = lorawanDevice.DisableFCntCheck
+
+	v3Dev.MACVersion = "MAC_V1_0_2"
+	v3Dev.PHYVersion = "PHY_V1_0_2_REV_B"
+
+	if lorawanDevice.AppKey.String() == "" {
+		v3Dev.SupportsJoin = false
+		v3Dev.MACSettings.Rx1Delay = "RX_DELAY_1"
+	} else {
+		v3Dev.SupportsJoin = true
+		if !lorawanDevice.AppSKey.IsEmpty() && !lorawanDevice.NwkSKey.IsEmpty() && !lorawanDevice.DevAddr.IsEmpty() {
+			v3Dev.MACState.CurrentParameters.Rx1Delay = "RX_DELAY_1"
+		}
+	}
```

Comment:
```suggestion
	v3Dev.SupportsJoin = lorawanDevice.AppKey.String() != ""
	v3Dev.MACSettings.Rx1Delay = "RX_DELAY_1"
```
1. Unfortunately Network Server does not support creating devices with MAC state at this point (it gets overwritten) - it's easy to fix and I will open an issue about it
2. Using RX Delay of 1 will give users more familiar experience, RX delay of 5 might in fact break some use cases, so IMO this is the correct approach. @johanstokking what do you think?

Reply:
Migrated device sessions have to respect the existing RX delay of 1 second, but I'm open to increasing it in the desired parameters.

Maybe I misunderstood your question. In that case, please clarify

Reply:
My suggestion is to explicitly use RX delay of 1 in MAC settings(i.e. always) for all migrated devices, see suggested diff.